File: models/project.js
const mongoose = require("mongoose");
const ObjectId = mongoose.Schema.Types.ObjectId;
const ENUM = require('./enums')

const projectSchema = new mongoose.Schema({
  id: ObjectId,
  userId: {
    type: String,
    required: true
  },
  name: { type: String },
  primaryMethod: {
    type: String,
    enum: ENUM.METHOD
  },
  projectStatus: {
    type: String,
    enum: ENUM.STATUS
  },
  dominantStat: {
    type: String,
    enum: ENUM.STATS
  },
  stats: {
    type: [String],
    enum: ENUM.STATS
  },
  projectUsages: [
    { type: mongoose.Schema.Types.ObjectId, ref: 'ProjectUsage' }
  ],
  totalTimeSpend:
  {
    type: Number,
    default: 0
  },
  dateAdded: {
    type: Date,
    default: Date.now
  }
});

const Project = mongoose.model("Project", projectSchema);
module.exports = Project;

File: Models/Project.cs
using System;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace ProjectsApi.Models
{
    [BsonIgnoreExtraElements]
    public class Project
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        [BsonElement("userId")]
        public string UserId { get; set; }

        [BsonElement("name")]
        public string Name { get; set; }

        [BsonElement("primaryMethod")]
        public METHOD? PrimaryMethod { get; set; }

        [BsonElement("projectStatus")]
        public STATUS? ProjectStatus { get; set; }

        [BsonElement("dominantStat")]
        public STATS? DominantStat { get; set; }

        [BsonElement("stats")]
        public STATS[] Stats { get; set; }

        [BsonRepresentation(BsonType.ObjectId)]
        [BsonElement("projectUsages")]
        public string[] ProjectUsages { get; set; }

        [BsonElement("totalTimeSpend")]
        public int TotalTimeSpend { get; set; }

        [BsonElement("dateAdded")]
        public DateTime DateAdded { get; set; }
    }
}